src/lib/plugins/clear.js

```diff
@@ -0,0 +1,204 @@
+module.exports.server = (serv, { version }) => {
+  const mcData = require('minecraft-data')(version)
+  const ChatMessage = require('prismarine-chat')(version)
+  const notUdf = x => x !== undefined
+  const playersString = players => players.length > 1 ? `${players.length} players` : `player ${players[0].username}`
+  serv.commands.add({
+    base: 'clear',
+    info: 'Clears items from player(s) inventory.',
+    usage: '/clear <player> <item> <maxCount>',
+    parse: /^((?:@|\w)+)?(?: (\w+))?(?: (\d+))?$/, // ex: /clear USERNAME_ stone 1
+    action (params, ctx) {
+      const isConsole = Object.entries(ctx).length === 0
+      const isOp = isConsole ? true : ctx.player.op
+      const [, playerSelector, removeBlockId, removeCount] = params
+      // errors
+      if (typeof params === 'string') return `${this.usage}: ${this.info}` // parsing failed
+      if (isConsole && playerSelector === undefined) return 'A player is required to run this command here' // console
+      if (!isOp && notUdf(playerSelector)) return 'Command not found'
+      if (notUdf(removeBlockId) && !mcData.itemsByName[removeBlockId]) return `Unknown item tag'${removeBlockId}'`
+      if (notUdf(removeCount) && isNaN(parseInt(removeCount))) return 'The number of blocks to remove given is invalid.'
+      // player(s) to execute on
+      let players
+      if (isConsole) {
+        players = [serv.getPlayer(playerSelector)]
+      } else if (notUdf(playerSelector)) {
+        players = Object.entries(serv.parseSelectorString(playerSelector, ctx.player.position, ctx.player.world)).map(p => p[1])
+      } else { // no input given (so default to executor)
+        players = [ctx.player]
+      }
+      if (players.some(player => player == null)) return "Player given doesn't exist"
```

Comment:
All this player selection logic looks like it could be put in its own function to be reused for other commands.

Reply:
done